x-pack/plugins/code/server/repository_service.ts

```diff
@@ -19,10 +21,24 @@ import {
 } from '../model';
 import { Logger } from './log';
 
-export type CloneProgressHandler = (progress: number, cloneProgress?: CloneProgress) => void;
+// Return false to stop the clone progress. Return true to keep going;
+export type CloneProgressHandler = (progress: number, cloneProgress?: CloneProgress) => boolean;
+export type UpdateProgressHandler = () => boolean;
 
+const GIT_FETCH_PROGRESS_CANCEL = -1;
+// TODO: Cannot directly access Git.Error.CODE.EUSER (-7). Investigate why.
+const NODEGIT_CALLBACK_RETURN_VALUE_ERROR = -7;
+const GIT_INDEXER_PROGRESS_CALLBACK_RETURN_VALUE_ERROR_MSG = `indexer progress callback returned ${GIT_FETCH_PROGRESS_CANCEL}`;
```

Comment:
is those text contructed by nodegit?

Reply:
constructed by nodegit, but no constant to reference for this case.